src/DelegateListOverview/DelegateListOverview.tsx

```diff
@@ -0,0 +1,378 @@
+
+    // handling of the most left pagination button (<<)
+    const handleClickFirst = () => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        alteredRequestOptions.PageNumber = 1;
+        setRequestDelegateOffersOptions(alteredRequestOptions);
+        handleLoadDelegates()
+    }
+
+    // Handling of the second pagination button (<)
+    const handleClickPrevious = () => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        alteredRequestOptions.PageNumber = requestDelegateOffersOptions.PageNumber - 1;
+        setRequestDelegateOffersOptions(alteredRequestOptions);
+        handleLoadDelegates()
+    }
+
+    // Handling of the number in between (*)
+    const handleClickNumber = (number: number) => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        alteredRequestOptions.PageNumber = number;
+        setRequestDelegateOffersOptions(alteredRequestOptions);
+        handleLoadDelegates()
+    }
+
+    // Handling of the second to last pagination button (>)
+    const handleClickNext = () => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        alteredRequestOptions.PageNumber = requestDelegateOffersOptions.PageNumber + 1;
+        setRequestDelegateOffersOptions(alteredRequestOptions);
+        handleLoadDelegates()
+    }
+
+    // Handling of the last pagination button (>>)
+    const handleClickLast = () => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        alteredRequestOptions.PageNumber = headers.TotalPages;
+        setRequestDelegateOffersOptions(alteredRequestOptions);
+        handleLoadDelegates()
+    }
+
+    // Check if button should be disabled or not
+    const isPreviousDisabled = (): boolean => {
+        return headers.HasPrevious === false;
+    }
+
+    // Check if button should be disabled or not
+    const isNextDisabled = (): boolean => {
+        return headers.HasNext === false;
+    }
+
+    // Apply filtering method. All fields are copied over to the {requestDelegateOffersOptions}. Then this object is being used to do the API Call.
+    const applyFiltering = () => {
+        let alteredRequestOptions = requestDelegateOffersOptions;
+        //reset pagination
+        alteredRequestOptions.PageNumber = 1
+
+        //apply filtering on these fields
+        alteredRequestOptions.SearchQuery = searchQ
+        alteredRequestOptions.RegionQuery = regionQ
+        alteredRequestOptions.MinPrice = minPrice
+        alteredRequestOptions.MaxPrice = maxPrice
+        alteredRequestOptions.MinAvailableForInMonth = minAvailableForInMonth
+        alteredRequestOptions.MaxAvailableForInMonth = maxAvailableForInMonth
+
+        setRequestDelegateOffersOptions(alteredRequestOptions)
+
+        handleLoadDelegates()
+    }
+
+    // Clears the filtering fields.
+    const clearFilteringFields = () => {
+        setSearchQ("")
+        setRegionQ("")
+        // @ts-ignore
+        setMaxPrice("")
```

Comment:
de numbers heb je @ts-ignore waarom heb je niet een nummer of undefined ingevuld